Python tools improvements

rivos-eblot opened the pull request.

rivos-eblot committed:
[ot] python/qemu: ot.util.present: move module from scripts/opentitan directory

Signed-off-by: Emmanuel Blot <eblot@rivosinc.com>

rivos-eblot committed:
[ot] python/qemu: ot.util.misc: add redent helper function

Signed-off-by: Emmanuel Blot <eblot@rivosinc.com>

rivos-eblot committed:
[ot] python/qemu: ot.devproxy: use an older Python syntax for union type hinting

Signed-off-by: Emmanuel Blot <eblot@rivosinc.com>

rivos-eblot committed:
[ot] scripts/opentitan: .pylintrc: remove yet another "too-many" warning

Signed-off-by: Emmanuel Blot <eblot@rivosinc.com>

rivos-eblot committed:
[ot] python/qemu: ot.util.misc: add retrieve_git_version helper

Signed-off-by: Emmanuel Blot <eblot@rivosinc.com>

rivos-eblot committed:
[ot] python/qemu: ot.pyot.executer: improve reporting of context exec failure

Signed-off-by: Emmanuel Blot <eblot@rivosinc.com>

rivos-eblot committed:
[ot] python/qemu: ot.pyot.executer: use specialized wrapper

QEMU executer should use QEMU wrapper.

Signed-off-by: Emmanuel Blot <eblot@rivosinc.com>

rivos-eblot committed:
[ot] python/qemu: ot.pyot.util: fix execution time computation

Some fail test may report no execution time.

Signed-off-by: Emmanuel Blot <eblot@rivosinc.com>

AlexJones0 commented on scripts/opentitan/present.py:
```diff
@@ -1,5 +1,3 @@
-#!/usr/bin/env python3
```
I understand removing this shebang because this is now a util module and not a script, but maybe it should be kept because this file still has an `if __name__ == "__main__":` for handling it being interpreted directly?

rivos-eblot replied:
The module is no longer an executable script, but it can still be directly invoked from the command line to run some self-test:

````sh
python3 python/qemu/ot/util/present.py [-v]      
````

Good point though as it made me find that I did an ugly error using `argparse` as the extension module for the ArgumentParser while trying to run it.

It is definitely not a good idea to name a custom Python module as an existing one, even it is part of a module hiearchy such as `ot.util.argparse`... I fixes this issue.

AlexJones0 commented on python/qemu/ot/util/misc.py:
```diff
@@ -193,10 +196,83 @@ def to_bool(value, permissive=True, prohibit_int=False):
+        gmo = re.match(r'^(?P<tag>.*)-(?P<dist>\d+)-g(?P<commit>[0-9a-f]+)'
+                       r'(?:-(?P<dirty>dirty))?$', descstr)
+        if not gmo:
+            raise ValueError('Unknown Git describe format')
```
Maybe this could log the output?

```suggestion
            raise ValueError(f'Unknown Git describe format: {descstr}')
```

rivos-eblot replied:
I'm not sure how to handle this since `ValueError` is caught and discarded, _i.e._ the message is more a comment that something that gets reported. There is no logger here, and printing out to stderr is quite intrusive.

AlexJones0 replied:
I see, would it not be better to do anyway in case code is added in the future that relies on this function but does not discard the error?

rivos-eblot replied:
It does not hurt adding it, but using it would require to modify the function itself since it is the function that discards the ValueError (line 266/267)

AlexJones0 replied:
Ah, sorry, I missed the context that this was in a `try: ... except: ...` when I replied. Then yes, I guess it is fine to leave it as is.

AlexJones0 reviewed (approved):
LGTM, thanks. I've left a couple of minor comments/suggestions.

rivos-eblot committed:
[ot] python/qemu: ot.util.argparse: rename as ot.util.arg

Using an existing Python module name is a very bad idea as it ends up
conflicting when Python interpreter looks up for modules.

Signed-off-by: Emmanuel Blot <eblot@rivosinc.com>
